src/main/java/com/crud/tasks/trello/client/TrelloClient.java

```diff
@@ -0,0 +1,53 @@
+package com.crud.tasks.trello.client;
+
+import com.crud.tasks.domain.TrelloBoardDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class TrelloClient {
+
+    @Value("${trello.api.endpoint.prod}")
+    private String trelloApiEndpoint;
+
+    @Value("${trello.app.key}")
+    private String trelloAppKey;
+
+    @Value("${trello.app.token}")
+    private String trelloToken;
+
+    @Value("${trello.app.username}")
+    private String trelloUserName;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    public List<TrelloBoardDto> getTrelloBoards() {
+
+
+        Optional<TrelloBoardDto[]> boardsResponse = Optional.ofNullable(restTemplate.getForObject(getUri(),
+                TrelloBoardDto[].class));
+
+        if (boardsResponse.isPresent()) {
+            return Arrays.asList(boardsResponse.get());
+        }
+        return new ArrayList<>();
```

Comment:
linie 40-43 mozna zapisac
`return boardsResponse.map(Arrays::asList).orElseGet(ArrayList::new);`
zwroc uwage ze wykonujesz funkcje map na optionalu, poniewaz optional to stream o ilosci elementow 0 lub 1.